notebooks/118-optimize-preprocessing/118-optimize-preprocessing.ipynb

```diff
@@ -0,0 +1,537 @@
+{
```

Comment:
<p>&gt;&gt;Therefore, in this case, target height / width should be specified as PreProcessSteps.resize( ResizeAlgorithm, destination_height, destination_width).</p><p> </p><p>this sentence is not clear for me, as I know there is no opportunity to specify target height/width in preprocessing API. it uses model input shape instead. e.g. if model has input shape [1, 3, 224, 224], resize will use 224x224 as output size.</p><p> </p><p>I mean different when speak about dynamic.</p><p> </p><p>set_shape([1, 577, 800, 3]) - this method declare that all processed data always  have this size. It is good for example, when you process video, because all frames has the same resolution, right?</p><p>But we can not say it about fitting just images. If you want process image with different size, it should be resized to 800x577 before submitting to model or preprocessing should be reconfigured and rebuild again for next resolution and it is for each image, which can be not convinient.</p><p> </p><p>There is another way as I know to describe tensor for this condition. </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/811/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Reply:
I'm not sure, is there mention about it in docs,
set_spatial_dynamic_shape() used for cases where the original image size is unknown and can be any instead of tensor set_shape

Reply:
thank you ! yes, it is not mention in overview or details, I found it late in https://docs.openvino.ai/latest/openvino_docs_OV_UG_Preprocess_Usecase_save.html .